### ISSUE TITLE
Failed loading mixer: /etc/mixers/quad_x.main.mix

### ISSUE THREAD

**Issue by stmoon** [PX4/Firmware#9582]
Hi,

When I reboot quad copter using MAV_CMD_PREFLIGHT_REBOOT_SHUTDOWN, ARM didn't work with safety button off sometimes.
In this case I should power off and on again to work well.
 
And I found that [init] Failed loading mixer: /etc/mixers/quad_x.main.mix
Why does the mixer not work?

I have used v.1.7.4beta firmware.

Thanks

```
ERROR: Could not start: fmu mode_pwm 
ERROR  [init] Failed appending mixer: /etc/mixers/pass.aux.mix 
ERROR: Could not start: fmu mode_pwm 
ERROR  [init] Failed loading mixer: /etc/mixers/quad_x.main.mix 
ERROR  [init] Failed appending mixer: /etc/mixers/pass.aux.mix 
ERROR: Could not start: fmu mode_pwm 
PX4IO update failed 
PX4IO not found 
ERROR  [init] Failed loading mixer: /etc/mixers/quad_x.main.mix 
ERROR  [init] Failed appending mixer: /etc/mixers/pass.aux.mix 
ERROR: Could not start: fmu mode_pwm 
PX4IO update failed 
PX4IO not found 
ERROR  [init] Failed loading mixer: /etc/mixers/quad_x.main.mix 
ERROR  [init] Failed appending mixer: /etc/mixers/pass.aux.mix 
ERROR: Could not start: fmu mode_pwm 
```

**Comment by ChrisHughes** [comment]
Same issue here, Pixfalcon and 1.7.4beta.

**Comment by stmoon** [comment]
In my case, pixhawk 2.1 and same firmware

2018년 6월 4일 (월) 오전 3:46, ChrisHughes <notifications@github.com>님이 작성:

**Comment by stmoon** [comment]
Could you please let me know the reason or some solution to fix it? 

Thanks.

**Comment by LorenzMeier** [comment]
There is not nearly enough context to solve this. Please provide a complete writeup what you were trying to do, which configuration you were using and what you did before the reboot (including complete shell output).

This issue will auto-close in 30 days without any further input. There seems nothing to be wrong with the core system as far as we can tell with the information provided.

**Comment by stmoon** [comment]
Thanks you @LorenzMeier ,

I just rebooted the PX4(pixhawk 2.1) using VEHICLE_CMD_PREFLIGHT_REBOOT_SHUTDOWN with param 1 using mavlink.
I have used firmware v1.7.4

I think that this problem happens once per 10 times try. and I found the bootlog.txt was written like this
```
ERROR  [init] Failed loading mixer: /etc/mixers/quad_x.main.mix 
```
and I found the error happen in the mixer in rc.interface.

```
if mixer load ${OUTPUT_DEV} ${MIXER_FILE}
```
Once the reboot is failed, next time the mixer does not working any more.
To solve this, I just should connect battery again for cold booting.

Thanks in advance

**Comment by stmoon** [comment]
Hi @LorenzMeier 

Finally, I found the error when running "mixer load"

```
WARN  [px4io] fmu sent: "0
S: 3 5 10000 10000 0 -10000 10000
M: 1
O: 10000 10000 0 -"
px4io mixer send error -1
ERROR [mixer] failed to load mixers from /etc/mixers/quad_x.main.mix
ERROR [mixer] failed to load mixer
ERROR  [init] Failed loading mixer: /etc/mixers/quad_x.main.mix, ret:0
```

It is cause by the code in the mixer
```
	/* Pass the buffer to the device */
	int ret = px4_ioctl(dev, MIXERIOCLOADBUF, (unsigned long)buf);

```

Could you explain me why sometimes the warm rebooting is not working?

Thanks in advance

Best regards,
SungTae Moon

**Comment by stmoon** [comment]
Is there someone to help me?

**Comment by v01d** [comment]
I'm trying HIL and I found this on the bootlog, wonder if it is related since I get five low frequency tones on boot and I also cannot get hil actuator controls to be non-zero

**Comment by stmoon** [comment]
Hi @v01d 

maybe yes,..  however I didn't find any solution until now. :-(

**Comment by sunnyroad** [comment]
I also meet same issue, when reboot with mavlink cmd PREFLIGHT_REBOOT_SHUTDOWN, occasionally it do not work, and then mixer load fail, then same such reboot not work  any more, only power off-on could solve it.
![image](https://user-images.githubusercontent.com/7277416/59001934-20716700-8843-11e9-98de-c1479143fb3a.png)
![image](https://user-images.githubusercontent.com/7277416/59001941-25ceb180-8843-11e9-83df-091d15086570.png)

Anyone can help this?

Best regards.

**Comment by julianoes** [comment]
@sunnyroad on what Pixhawk is this? And does this happen when you reboot when armed or disarmed?

**Comment by sunnyroad** [comment]
> @sunnyroad on what Pixhawk is this? And does this happen when you reboot when armed or disarmed?

Pixhawk2,  actually issue happen when reboot FCU via command "PREFLIGHT_REBOOT_SHUTDOWN", so after that, when we send armed command, it also respond armed success but no pwm output on each FCU's PWM_OUT IO(always zero output).

**Comment by sunnyroad** [comment]
Hi, julianoes
   do you have any findings or fixs of this issue?

**Comment by julianoes** [comment]
Does this issue still exist in v1.9?

**Comment by sunnyroad** [comment]
> Does this issue still exist in v1.9?

Hi, we do not use v1.9,  is any related fix or enhances for this issue related after v1.8.2?

**Comment by julianoes** [comment]
I don't think anyone really has the time to fix issues from before v1.9. If you find a way to reproduce it on v1.10, re-open this issue.

Or if you find a fix, we'd be happy to see a pull request, thanks!
